synapse/storage/databases/state/store.py

```diff
@@ -404,6 +404,9 @@ async def store_state_group(
     ) -> int:
         """Store a new set of state, returning a newly assigned state group.
 
+        If `current_state_ids` is None then `prev_group` and `delta_ids` must
+        not be None.
```

Comment:
What you've written is true and complete, but slightly challenging to grok.

```suggestion
        At least one of `current_state_ids` and `prev_group` must be provided. Whenever
        `prev_group` is not None, `delta_ids` must also not be None.
```